Updated examples/pytorch to disable istio sidecar injection

jdcfd opened the pull request:
<!--  Thanks for sending a pull request! Here are some tips for you:
1. If this is your first time, check our contributor guidelines: https://www.kubeflow.org/docs/about/contributing
2. To know more about Training Operator, check the developer guide:
    https://github.com/kubeflow/training-operator/blob/master/docs/development/developer_guide.md
3. If you want *faster* PR reviews, check how: https://git.k8s.io/community/contributors/guide/pull-requests.md#best-practices-for-faster-reviews
-->

**What this PR does / why we need it**:
Examples do not work because of istio sidecar injection causing trouble
**Which issue(s) this PR fixes** _(optional, in `Fixes #<issue number>, #<issue number>, ...` format, will close the issue(s) when PR gets merged)_: 
Fixes #1637

**Checklist:**

- [x] [Docs](https://www.kubeflow.org/docs/components/training/) included if any changes are user facing (PR https://github.com/kubeflow/website/pull/3686)

jdcfd commented:
/check-cla

jdcfd commented:
Already updated cla... I had submitted with the wrong e-mail...

/check-cla

tenzen-y reviewed (commented):
Thank you!
/lgtm
/approve

/assign @kubeflow/wg-training-leads 
for approval CI.

tenzen-y commented:
/hold

andreyvelich commented:
Thank you for creating this PR @jdcfd!
@tenzen-y @johnugeorge I remember previously we had discussion that we should not add Istio annotation to the Examples to not confuse users: https://github.com/kubeflow/katib/issues/1359#issue-723619881
At least we did it for Katib.
Should we follow the same model for Training Operator examples, but just update the documentation ?

tenzen-y commented:
> Thank you for creating this PR @jdcfd! @tenzen-y @johnugeorge I remember previously we had discussion that we should not add Istio annotation to the Examples to not confuse users: [kubeflow/katib#1359 (comment)](https://github.com/kubeflow/katib/issues/1359#issue-723619881) At least we did it for Katib. Should we follow the same model for Training Operator examples, and just update the documentation ?

@andreyvelich I'm ok with either way. I think that the inconsistent examples and docs brought confusions :(
So, if we remove the istio annotation, we should remove the annotation from all examples and add documentation as you say.

jdcfd commented:
Hi all, thanks for the feedback. I have to admit this is all new to me, even the process of submitting a PR to an open-source project so bear with me. 

From what I understood, it is not always necessary to add the Istio annotation, therefore, the user should do it manually if necessary. That was the path you took for katib. In that case, one option is to remove annotation from all examples and add a comment in `examples/pytorch/README.md` . 

Is this the same with all other training operators? or only pytorch?

tenzen-y commented:
> Hi all, thanks for the feedback. I have to admit this is all new to me, even the process of submitting a PR to an open-source project so bear with me.
> 
> From what I understood, it is not always necessary to add the Istio annotation, therefore, the user should do it manually if necessary. That was the path you took for katib. In that case, one option is to remove annotation from all examples and add a comment in `examples/pytorch/README.md` .
> 
> Is this the same with all other training operators? or only pytorch?

Oh, I see. Welcome to the kubeflow community :)
Yes, we should remove the istio annotations from all training operators.

andreyvelich commented:
Welcome to the community @jdcfd!
Yes, @tenzen-y is right, we can remove istio annotations from all examples.
That would be great if you could submit PR in [the Kubeflow Website repo](https://github.com/kubeflow/website) to explain how users can disable Istio for every job.
Maybe we can follow the TFJob guide and update other CRs accordingly: https://www.kubeflow.org/docs/components/training/tftraining/#what-is-tfjob

tenzen-y commented:
/lgtm cancel
/approve cancel

tenzen-y commented:
/approve cancel

jdcfd commented:
@tenzen-y @andreyvelich I have removed the istio annotations from the examples, added some comments to the README file for PyTorch jobs and submitted a PR to the kubeflow/website repo: https://github.com/kubeflow/website/pull/3686

andreyvelich commented on examples/pytorch/cpu-demo/demo.yaml:
```diff
@@ -13,7 +13,7 @@ spec:
             - name: pytorch
               image: pytorch-cpu:py3.8
-              imagePullPolicy: Always
+              imagePullPolicy: IfNotPresent
```
You can remove this imagePullPolicy, since IfNotPresent is set by default in Kubernetes container.

jdcfd commented:
@tenzen-y I removed the istio annotations as discussed. I also submitted a PR to the documentation website repo.

andreyvelich reviewed (approved):
Thank you for your contribution @jdcfd 🎉 
/lgtm
/approve

andreyvelich commented:
/hold cancel
